test/integration/daemonset/daemonset_test.go

```diff
@@ -875,10 +892,9 @@ func TestLaunchWithHashCollision(t *testing.T) {
 
 	// Make an update of the DaemonSet which we know will create a hash collision when
 	// the next ControllerRevision is created.
-	_, err = dsClient.Update(ds)
-	if err != nil {
-		t.Fatalf("Failed to update DaemonSet: %v", err)
-	}
+	updateDS(t, dsClient, ds.Name, func(updateDS *apps.DaemonSet) {
+		updateDS.Spec.Template.Spec.TerminationGracePeriodSeconds = &one
```

Comment:
Combine L874 with this

Reply:
The change to `TerminationGracePeriodSeconds` on line 874 is needed to compute the hash for the controller revision that will create the hash collision when the daemonset is updated. I'm not sure there is an easy way to combine this since the controller revision will need to be created before the call to `Update`.